src/lib.rs

```diff
@@ -475,8 +509,17 @@ impl<T> ops::DerefMut for Ref<'_, T> {
 impl<T> Drop for Ref<'_, T> {
     #[inline]
     fn drop(&mut self) {
-        test_println!("drop Ref<{}>", core::any::type_name::<T>());
-        test_dbg!(self.slot.state.store(test_dbg!(self.new_state), Release));
+        if self.is_pop {
+            test_println!("drop Ref<{}> (pop)", core::any::type_name::<T>());
+            test_dbg!(self.slot.has_reader.store(test_dbg!(false), SeqCst));
+        } else {
+            test_println!(
+                "drop Ref<{}> (push), new_state = {}",
+                core::any::type_name::<T>(),
+                self.new_state
+            );
+            test_dbg!(self.slot.state.store(test_dbg!(self.new_state), Release));
```

Comment:
if we took my suggestion of storing the reader bit in the `state` field, the `Drop` impl could be simplified, and `Ref` could be one word smaller, because we would either store a `new_state` that advances the generation, or one that clears the reader bit.